### PR TITLE
KUZ-673 Add getConfig route into admin controller

### PR DESCRIPTION
**Pull request by j33f**

### CODE REVIEW

File: lib/api/controllers/adminController.js [hunk 1]
```diff
@@ -102,6 +102,49 @@ function AdminController(kuzzle) {
...
+  this.getConfig = requestObject => {
+    var modifiedRequestObject = null;
+
+    return kuzzle.pluginsManager.trigger('data:beforeGetConfig', requestObject)
+      .then(newRequestObject => {
+        var pluginsConfig = {};
+
+        modifiedRequestObject = newRequestObject;
+
+        return kuzzle.internalEngine
```
Comment (stafyniaksacha) [inline lib/api/controllers/adminController.js]: You have exactly what you expected in `kuzzle.pluginsManager.plugins`.

You could simply do: 
```javascript
this.getConfig = requestObject => {
  var modifiedRequestObject = null;

  return kuzzle.pluginsManager.trigger('data:beforeGetConfig', requestObject)
    .then(newRequestObject => {
      var pluginsConfig = {};

      modifiedRequestObject = newRequestObject;

      return Promise.resolve({
          kuzzle: kuzzle.config,
          plugins: kuzzle.pluginsManager.plugins
        });
     })
    .then(response => kuzzle.pluginsManager.trigger('data:afterGetConfig', new ResponseObject(modifiedRequestObject, response)));
  };
```

Comment (j33f) [inline lib/api/controllers/adminController.js]: right, but we need to cleanup each plugin context to remove the duplicated kuzzle config in it

Comment (stafyniaksacha) [inline lib/api/controllers/adminController.js]: Hum, it's a good question.

In `kuzzle.pluginsManager.plugins` we have the configuration context of the plugin. Not each plugin require the kuzzle's config.

Comment (j33f) [inline lib/api/controllers/adminController.js]: some of them have it included, its ok (see last commit)

File: lib/config/httpRoutes.js [hunk 2]
```diff
@@ -3,6 +3,31 @@ module.exports = [
   {verb: 'get', url: '/_logout', controller: 'auth', action: 'logout'},
   {verb: 'get', url: '/_getLastStats', controller: 'admin', action: 'getLastStats'},
   {verb: 'get', url: '/_getAllStats', controller: 'admin', action: 'getAllStats'},
+  {verb: 'get', url: '/_getConfig', controller: 'admin', action: 'getConfig', infos: {
+    description: 'Retrieve the current Kuzzle configuration',
+    parameters: [],
+    produces: ['application/json'],
+    responses: {
+      '200': {
+        description: 'Everything went good, the configuration have been retrieved',
```
Comment (scottinet) [inline lib/config/httpRoutes.js]: -good +fine
-have +has

Other proposition: `Successfully retrieved Kuzzle configuration`

File: lib/config/httpRoutes.js [hunk 3]
```diff
@@ -3,6 +3,31 @@ module.exports = [
...
+        schema: {
+          type: 'object',
+          properties: {
+            action: {type: 'string', description: 'The action into Kuzzle'},
```
Comment (scottinet) [inline lib/config/httpRoutes.js]: Better description: `Kuzzle controller's action to perform`

File: lib/config/httpRoutes.js [hunk 4]
```diff
@@ -3,6 +3,31 @@ module.exports = [
...
+            controller: {type: 'string', description: 'The controller into Kuzzle'},
```
Comment (scottinet) [inline lib/config/httpRoutes.js]: Better description: `Kuzzle controller to invoke`

File: lib/config/httpRoutes.js [hunk 5]
```diff
@@ -3,6 +3,31 @@ module.exports = [
...
+            error: {type: 'null', description: 'The error object or null'},
```
Comment (scottinet) [inline lib/config/httpRoutes.js]: -The +An

File: lib/config/httpRoutes.js [hunk 6]
```diff
@@ -3,6 +3,31 @@ module.exports = [
...
+            metadata: {type: 'object', description: 'The Metadatas'},
```
Comment (scottinet) [inline lib/config/httpRoutes.js]: Better description: `Request metadata` (no -s, "data" is already a plural)

File: lib/config/httpRoutes.js [hunk 7]
```diff
@@ -3,6 +3,31 @@ module.exports = [
...
+            requestId: {type: 'object', description: 'The Request ID'},
```
Comment (scottinet) [inline lib/config/httpRoutes.js]: Better description: `Request unique identifier`

File: lib/api/controllers/adminController.js [hunk 8]
```diff
@@ -102,6 +102,44 @@ function AdminController(kuzzle) {
...
+    return kuzzle.pluginsManager.trigger('data:beforeGetConfig', requestObject)
+      .then(newRequestObject => {
+        var
+          pluginsConfig = {},
+          response;
+
+        modifiedRequestObject = newRequestObject;
+
+        _.forEach(kuzzle.pluginsManager.plugins, (object, id) => {
+          pluginsConfig[id] = _.assignIn(object);
```
Comment (j33f) [inline lib/api/controllers/adminController.js]: eeeew, typo...

**Comment by j33f** [comment]
@scottinet @stafyniaksacha all requested changes are done

**Pull request merged**